cpp_linter/run.py

```diff
@@ -678,30 +589,34 @@ def make_annotations(
     )
     for advice, file in zip(GlobalParser.format_advice, files):
         line_filter = cast(List[int], range_of_changed_lines(file, lines_changed_only))
-        if advice.replaced_lines and file_annotations:
+        if advice.replaced_lines:
             output = advice.log_command(style, line_filter)
             if output is not None:
-                log_commander.info(output)
+                if file_annotations:
+                    log_commander.info(output)
                 count += 1
     for note in GlobalParser.tidy_notes:
         if lines_changed_only:
-            filename = note.filename.replace("\\", "/")
+            filename = note.filename.replace("\\", "/").lstrip("/")
             line_filter = []
             for file in files:
+                print(filename, "?=", file["filename"])
                 if filename == file["filename"]:
                     line_filter = cast(
                         List[int], range_of_changed_lines(file, lines_changed_only)
                     )
                     break
-            else:
+            else: # filename match not found; treat line_filter as empty list
                 continue
```

Comment:
I _think_ we should error here if filename was not matched, but I can't remember why I had previously set it to `continue`. If no filename match was found then, clang-tidy notes are not filtered by line changes, which seems ok for fallback behavior.

Reply:
Maybe if filename match wasn't found then it is possibly from a file in C/C++ std libs. Probably better to leave as is.

Reply:
It is not a syntax error; see [grammar of python `for` loop](https://docs.python.org/3/reference/compound_stmts.html#the-for-statement). The `else` block is only triggered if the `for` loop never `break`s:
```python
for i in range(3):
    if i > 3:
        break
else:
    # usually this is where errors are dealt with
    print("`i` was never greater than 3")
```